service/src/integrationTest/kotlin/fi/espoo/evaka/messaging/MessageIntegrationTest.kt

```diff
@@ -1156,6 +1204,150 @@ class MessageIntegrationTest : FullApplicationTest(resetDbBeforeEach = true) {
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_6.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_8.id),
+                ),
+            filters =
+                MessageController.PostMessageFilters(
+                    serviceNeedIds =
+                        listOf(snDefaultPartDayDaycare.id, snDefaultFiveYearOldsPartDayDaycare.id)
+                ),
+            user = messager,
+            now = sendTime
+        )
+
+        db.read {
+            assertEquals(
+                3,
+                @Suppress("DEPRECATION")
+                it.createQuery("SELECT COUNT(id) FROM message_recipients").exactlyOne<Int>()
+            )
+        }
+    }
+
+    @Test
+    fun `recipient list can be filtered 3`() {
+        postNewThread(
+            title = "Vappu",
+            message = "Vappuna paistaa aurinko",
+            messageType = MessageType.BULLETIN,
+            sender = messagerAccount,
+            recipients =
+                listOf(
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_1.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_3.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_4.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_6.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_8.id),
+                ),
+            filters =
+                MessageController.PostMessageFilters(
+                    shiftCare = true,
+                    intermittentShiftCare = true
+                ),
+            user = messager,
+            now = sendTime
+        )
+
+        db.read {
+            assertEquals(
+                3,
+                @Suppress("DEPRECATION")
+                it.createQuery("SELECT COUNT(id) FROM message_recipients").exactlyOne<Int>()
+            )
+        }
+    }
+
+    @Test
+    fun `recipient list can be filtered 4`() {
+        postNewThread(
+            title = "Vappu",
+            message = "Vappuna paistaa aurinko",
+            messageType = MessageType.BULLETIN,
+            sender = messagerAccount,
+            recipients =
+                listOf(
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_1.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_3.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_4.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_6.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_8.id),
+                ),
+            filters = MessageController.PostMessageFilters(familyDaycare = true),
+            user = messager,
+            now = sendTime
+        )
+
+        db.read {
+            assertEquals(
+                2,
+                @Suppress("DEPRECATION")
```

Comment:
Use the new API instead of the deprecated old one: `it.createQuery { sql(...) }`